server/models/Trips.js

```diff
@@ -79,6 +87,17 @@ class Trips {
 		return this.result;
 	}
 
+	static async viewSingleActivetrip(tripId) {
+		// eslint-disable-next-line radix
+		const id = parseInt(tripId);
+		const obj = db.find(o => o.id === id && o.status === 'active');
+		if (!obj) {
+			return false;
+		}
+		this.result = obj;
+		return this.result;
```

Comment:
same here, the keyword "static" dissociates a method from an instance of a class. In such a case, "this" would be undefined

Reply:
i have pushed changes where i have updated my methods.